roles/openshift_cluster_monitoring_operator/defaults/main.yml

```diff
@@ -26,16 +21,16 @@ l_openshift_cluster_monitoring_image_dicts:
     kube_rbac_proxy: quay.io/coreos/kube-rbac-proxy
     oauth_proxy: openshift/oauth-proxy
   openshift-enterprise:
-    prometheus_operator: "{{l_openshift_cluster_monitoring_operator_ocp_image_registry}}ose-prometheus-operator"
-    prometheus: "{{l_openshift_cluster_monitoring_operator_ocp_image_registry}}prometheus"
-    alertmanager: "{{l_openshift_cluster_monitoring_operator_ocp_image_registry}}prometheus-alertmanager"
-    node_exporter: "{{l_openshift_cluster_monitoring_operator_ocp_image_registry}}prometheus-node-exporter"
-    prometheus_config_reloader: "{{l_openshift_cluster_monitoring_operator_ocp_image_registry}}ose-prometheus-config-reloader"
-    configmap_reloader: "{{l_openshift_cluster_monitoring_operator_ocp_image_registry}}ose-configmap-reload"
-    grafana: "{{l_openshift_cluster_monitoring_operator_ocp_image_registry}}ose-grafana"
-    kube_state_metrics: "{{l_openshift_cluster_monitoring_operator_ocp_image_registry}}ose-kube-state-metrics"
-    kube_rbac_proxy: "{{l_openshift_cluster_monitoring_operator_ocp_image_registry}}ose-kube-rbac-proxy"
-    oauth_proxy: "{{l_openshift_cluster_monitoring_operator_ocp_image_registry}}oauth-proxy"
+    prometheus_operator: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}:${version}' | regex_escape, 'prometheus-operator') }}"
+    prometheus: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('ose-${component}:${version}' | regex_escape, 'prometheus') }}"
+    alertmanager: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('ose-${component}:${version}' | regex_escape, 'prometheus-alertmanager') }}"
+    node_exporter: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('ose-${component}:${version}' | regex_escape, 'prometheus-node-exporter') }}"
+    prometheus_config_reloader: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}:${version}' | regex_escape, 'prometheus-config-reloader') }}"
+    configmap_reloader: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}:${version}' | regex_escape, 'configmap-reload') }}"
+    grafana: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}:${version}' | regex_escape, 'grafana') }}"
+    kube_state_metrics: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}:${version}' | regex_escape, 'kube-state-metrics') }}"
+    kube_rbac_proxy: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}:${version}' | regex_escape, 'kube-rbac-proxy') }}"
+    oauth_proxy: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}:${version}' | regex_escape, 'oauth-proxy') }}"
```

Comment:
I think the oauth_proxy image needs to use `regex_replace('ose-${component}:${version}'`